docs/design/PicoD-Design.md

```diff
@@ -0,0 +1,674 @@
+
+```python
+
+from agentcube import CodeInterpreterClient
+
+# Create a CodeInterpreterClient instance
+code_interpreter = CodeInterpreterClient(
+    ttl=3600,  # Time-to-live in seconds
+    image="sandbox:latest",  # Container image to use
+)
+
+try:
+    # Step 1: Upload dependencies file (WriteFile API)
+    code_interpreter.write_file(
+        content="pandas\nnumpy\nscikit-learn\nmatplotlib",
+        remote_path="/workspace/requirements.txt"
+    )
+
+    # Step 2: Install dependencies (Execute API)
+    code_interpreter.execute_command("pip install -r /workspace/requirements.txt")
+
+    # Step 3: Upload training data (WriteFile API)
+    code_interpreter.upload_file(
+        local_path="./data/train.csv",
+        remote_path="/workspace/train.csv"
+    )
+
+    # Step 4: Train model (Execute API)
+    training_code = """
+    import pandas as pd
+    from sklearn.linear_model import LinearRegression
+    import pickle
+
+    df = pd.read_csv('/workspace/train.csv')
+    X, y = df[['feature1', 'feature2']], df['target']
+
+    model = LinearRegression().fit(X, y)
+    pickle.dump(model, open('/workspace/model.pkl', 'wb'))
+    print(f'Model R² score: {model.score(X, y):.4f}')
+    """
+    result = code_interpreter.run_code("python", training_code)
+
+    print(result)
+
+    # Step 5: Download trained model (ReadFile API)
+    code_interpreter.download_file(
+        remote_path="/workspace/model.pkl",
+        local_path="./models/model.pkl"
+    )
+
+    print("Workflow completed successfully!")
+
+finally:
+    code_interpreter.stop()
+
+```
+
+**API Calls Flow**:
+
+1. **POST /tools/code-interpreter/files**: Upload requirements.txt via multipart/form-data or JSON base64
+2. **POST /tools/code-interpreter/execute**: Install dependencies via pip command
+3. **POST /tools/code-interpreter/files**: Upload training data CSV file
+4. **POST /tools/code-interpreter/execute**: Run Python training code that processes data and trains model
+5. **GET /tools/code-interpreter/files/{path}**: Download trained model
+
+All operations use standard HTTP requests with token authentication in Authorization header.
+
+## Design Principles
+
+PicoD follows REST API best practices for simplicity and broad compatibility:
+
+### Architecture Patterns
+
+- **RESTful Design**: Resource-oriented architecture with standard HTTP methods
+- **JSON Payloads**: Human-readable request/response format
+- **Stateless**: Each request contains all necessary information
+- **Token Authentication**: Simple bearer token in Authorization header
+- **Standard HTTP Status Codes**: 200 OK, 400 Bad Request, 401 Unauthorized, 404 Not Found, 500 Internal Server Error
+
+### Core API Endpoints
+
+1. **POST /tools/code-interpreter/execute** - Execute commands
+2. **POST /tools/code-interpreter/files** - Upload files
+3. **GET /tools/code-interpreter/files/{path}** - Download files
+4. **GET /tools/code-interpreter/health** - Health check endpoint
```

Comment:
should simplify by removing `code-interpreter`, maybe
 `/v1/execute` 
 `/v1/files`